tests/posix/socket/enc/enc.c

```diff
@@ -0,0 +1,187 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* Licensed under the MIT License. */
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/time.h>
+
+// enclave.h must come before socket.h
+#include <openenclave/corelibc/arpa/inet.h>
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/netinet/in.h>
+#include <openenclave/corelibc/sys/socket.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/tests.h>
+#include <unistd.h>
+
+#include <socket_test_t.h>
+#include <stdio.h>
+#include <string.h>
+
+static void _initialize()
+{
+    OE_TEST(oe_load_module_host_socket_interface() == OE_OK);
+
+    {
+        char buf[1024] = {0};
+        OE_TEST(gethostname(buf, sizeof(buf)) == 0);
+        printf("hostname=%s\n", buf);
+        OE_TEST(strlen(buf) > 0);
+    }
+
+    {
+        char buf[1024] = {0};
+        OE_TEST(getdomainname(buf, sizeof(buf)) == 0);
+        printf("domainname=%s\n", buf);
+        //  OE_TEST(strlen(buf) > 0); Always fails on jenkins
+    }
+}
+
+/* This client connects to an echo server, sends a text message,
+ * and outputs the text reply.
+ */
+int ecall_run_client(char* recv_buff, ssize_t* recv_buff_len)
+{
+    _initialize();
+    int sockfd = 0;
+    ssize_t n = 0;
+    size_t buff_len = (size_t)*recv_buff_len;
+    struct oe_sockaddr_in serv_addr = {0};
+
+    memset(recv_buff, '0', buff_len);
+    printf("create socket\n");
+    if ((sockfd = oe_socket(OE_AF_INET, OE_SOCK_STREAM, 0)) < 0)
+    {
```

Comment:
Do we have test cases for SOCK_DGRAM socket?

Reply:
There are no tests for datagram sockets so I added issue #1872 so we can add them later.